pkg/config/controllerconfig.go

```diff
@@ -85,23 +89,15 @@ func (c *ControllerConfig) GRPCClientOptions() ([]option.ClientOption, error) {
 	if c.UserAgent != "" {
 		opts = append(opts, option.WithUserAgent(c.UserAgent))
 	}
-	if c.HTTPClient != nil {
-		// TODO: Set UserAgent in this scenario (error is: WithHTTPClient is incompatible with gRPC dial options)
-
-		httpClient := &http.Client{}
-		*httpClient = *c.HTTPClient
-		httpClient.Transport = &optionsRoundTripper{
-			config: *c,
-			inner:  c.HTTPClient.Transport,
-		}
-		opts = append(opts, option.WithHTTPClient(httpClient))
-	}
 	if c.UserProjectOverride && c.BillingProject != "" {
 		opts = append(opts, option.WithQuotaProject(c.BillingProject))
 	}
 	if c.GCPTokenSource != nil {
 		opts = append(opts, option.WithTokenSource(c.GCPTokenSource))
 	}
+	if c.GRPCUnaryClientInterceptor != nil {
+		opts = append(opts, option.WithGRPCDialOption(grpc.WithUnaryInterceptor(c.GRPCUnaryClientInterceptor)))
```

Comment:
can we provide a default UnaryClientInterceptor?

Reply:
Do we need one? I thought it's fine to use an empty interceptor (and not intercept anything) for production workflow (not in tests).

Reply:
Okay, as long as it proves working in k8s cluster against real GCP. IIRC, the MockGCP overrides this field for gRPC so the mock result may not be good to test this.

Reply:
Hmmm, I tested this with `hack/record-gcp`. Not sure actually if it will work in real cluster, without the interceptor.

Reply:
I think, based on https://github.com/grpc/grpc-go/blob/3f95b38ded016ebf32507fc7cb6baeb2f15aef59/interceptor.go#L28-L43, it should be optional.